torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -169,6 +169,26 @@ bool ShardingUtil::SetHloSharding(LoweringContext* lowering_ctx) {
   return is_sharded;
 }
 
+ShardingUtil::ShardingType ShardingUtil::GetShardingType(
+    xla::OpSharding& sharding) {
+  switch (sharding.type()) {
+    case xla::OpSharding::REPLICATED:
+      return ShardingType::REPLICATED;
+    case xla::OpSharding::MAXIMAL:
+      return ShardingType::MAXIMAL;
+    case xla::OpSharding::TUPLE:
+      return ShardingType::TUPLE;
+    case xla::OpSharding::OTHER:
+      // OTHER sharding can indicate either PARTIAL or TILED sharding.
+      return sharding.replicate_on_last_tile_dim() ? ShardingType::PARTIAL
```

Comment:
This seems pretty hacky. But I guess we don't have other ways round?

Reply:
My understanding is that we distinguish partial replication as a different sharding type whereas XLA treats partial and tiled as the same type `OTHER`. @yeounoh could you confirm?

Reply:
Yes, this is actually the correct way, the compiler treats TILED and PARTIAL the same as the OTHER type. The differences between the two would be how the tile shards are assigned to difference devices.